pandas/io/json/_json.py

```diff
@@ -633,19 +637,20 @@ def read_json(
         numpy=numpy,
         precise_float=precise_float,
         date_unit=date_unit,
-        encoding=encoding,
+        encoding=ioargs.encoding,
         lines=lines,
         chunksize=chunksize,
-        compression=compression,
+        compression=ioargs.compression,
         nrows=nrows,
     )
 
     if chunksize:
         return json_reader
 
     result = json_reader.read()
-    if should_close:
-        filepath_or_buffer.close()
+    if ioargs.should_close:
+        assert isinstance(ioargs.filepath_or_buffer, (IO, IOBase))
```

Comment:
it might be worth it to have a `is_closeable` tuple in `pandas._typing`?

Reply:
yes happy to add this to make these statements more concise